File: hotel-fronted/src/api/room.js
/**
 * Room API —— 对应后端 RoomController (/api/v1/rooms)
 */
import request from './index'

/** 房态地图：GET /rooms/status-map，返回 { available: [], occupied: [], dirty: [], maintenance: [] } */
export function getStatusMap() {
  return request.get('/rooms/status-map')
}

/** 房间详情（含房型信息）：GET /rooms/:id */
export function getRoomDetail(id) {
  return request.get(`/rooms/${id}`)
}

/** 可售房间列表，可选按房型过滤：GET /rooms/available?roomTypeId= */
export function getAvailableRooms(roomTypeId) {
  return request.get('/rooms/available', { params: { roomTypeId } })
}

File: hotel-backend/src/main/java/com/seeleaf/hotel/entity/Room.java
package com.seeleaf.hotel.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("room")
public class Room {

    @TableId
    private Long id;

    private String roomNumber;

    private Integer floor;

    private Long roomTypeId;

    /** available(可售) / occupied(已入住) / dirty(待清洁) / maintenance(维修) */
    private String status;

    /** 乐观锁版本号，防止并发操作导致房态冲突 */
    @Version
    private Integer version;

    @TableLogic
    private Integer isDeleted;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    /** 非数据库字段：由 RoomServiceImpl.getRoomDetail 手动填充 */
    @TableField(exist = false)
    private RoomType roomType;
}
